docs/migration-guide-v2.md

```diff
@@ -0,0 +1,276 @@
+# Parser v1 to v2 Migration Guide
+
+- The document's AsyncAPI API has been rewritten from old to new supporting the [Intent API](https://github.com/asyncapi/parser-api) - there is a function that converts the new API to the old one for backward compatibility. 
+- Package source code has been rewritten to the TypeScript alongside with rewritten from CJS (CommonJS) to ESM (EcmaScript) modules.
+- The functionality of the Parser was wrapped with the class. This means that from now on it's possible to have a dozen other Parser instances with different configurations.
+- Validation is powered by [Spectral](https://github.com/stoplightio/spectral). This allows better validation, checking not only syntax errors, but also informing about good practices, such as defining `operationID` field etc. In addition, Spectral uses a different references resolver, which causes changes in defining custom resolvers.
+- AsyncAPI document validation (without parsing) is available via a separate method - `.validate()`.
+- The function `.parse()` returns parsed AsyncAPi `document` and possible `diagnostics` based on validation configuration.  
+- The function `parseFromUrl()` has been deprecated in favour of `fromURL` standalone function.
+- The Custom Schema Parser interface has been changed with an additional function `validate()`.
+- Stringify and unstringify the parsed document is now a separate functionality.
+- Stringify and unstringify the parsed document is now a separate functionality.
+- Package is fully [tree-shakable](https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking).
```

Comment:
```suggestion
```

This is already described in the Typescript and ESM section.

Reply:
Just because some lib is written with ESM does not mean it is fully tree-shakable :) it is better to list it here so that people do not have to look in other subsections.

Reply:
Ok, I changed second point in the list to the:

> Source code has been rewritten in TypeScript and ESM. This makes the package fully [tree-shakable](https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking). [Read more](#typescript-and-esm).

What do you think?

Reply:
Yeah, that's much better. Thanks!